metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/index/entity/v3/AspectMappingBuilder.java

```diff
@@ -0,0 +1,300 @@
+package com.linkedin.metadata.search.elasticsearch.index.entity.v3;
+
+import static com.linkedin.metadata.Constants.STRUCTURED_PROPERTIES_ASPECT_NAME;
+import static com.linkedin.metadata.models.annotation.SearchableAnnotation.OBJECT_FIELD_TYPES;
+import static com.linkedin.metadata.search.utils.ESUtils.ALIAS_FIELD_TYPE;
+import static com.linkedin.metadata.search.utils.ESUtils.PATH;
+import static com.linkedin.metadata.search.utils.ESUtils.PROPERTIES;
+import static com.linkedin.metadata.search.utils.ESUtils.TYPE;
+
+import com.google.common.collect.ImmutableMap;
+import com.linkedin.data.schema.DataSchema;
+import com.linkedin.metadata.models.EntitySpec;
+import com.linkedin.metadata.models.annotation.SearchableAnnotation.FieldType;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Builder class for creating aspect-based field mappings. Handles the creation of mappings for
+ * searchable fields within aspects.
+ */
+@Slf4j
+public class AspectMappingBuilder {
+
+  /**
+   * Creates mappings for all aspects in an entity spec.
+   *
+   * @param entitySpec the entity spec to process
+   * @param fieldNameConflicts map of field names that have conflicts
+   * @return map of aspect mappings
+   */
+  public static Map<String, Object> createAspectMappings(
+      @Nonnull EntitySpec entitySpec,
+      @Nullable Map<String, Set<String>> fieldNameConflicts,
+      @Nullable Map<String, Set<String>> fieldNameAliasConflicts) {
+
+    Map<String, Object> aspectsMappings = new HashMap<>();
+
+    // Process searchable fields - they will be grouped under _aspects.aspectName
+    // Exception: structuredProperties aspect remains at root level
+    entitySpec
+        .getAspectSpecs()
+        .forEach(
+            aspectSpec -> {
+              String aspectName = aspectSpec.getName();
+
+              if (STRUCTURED_PROPERTIES_ASPECT_NAME.equals(aspectName)) {
```

Comment:
Seems like it shouldn't even enter down to this point if it's handled differently